### DISCUSSION TITLE
Applicable atomic file types for each model

### DISCUSSION THREAD

**Discussion by hlyu9** [RUCAIBox/RecBole#1189] [Q&A] [answered]
Hi, thank you so much for this excellent package! I have a noob question: how do we tell what type of atomic file could be applied for a model? For instance, I would like to train BERT4Rec with a dataset that can be divided into .inter, .user, and .item atomic files. While the official guide only specified that .inter is required for sequential models, I am not sure if I could also include .item and .user files to provide more information.

十分感谢各位大佬的开源！我想请教一个小白问题：我们怎么知道某个模型可以用哪些atomic file呢？我现在想用一个包含了.item, .user, .inter的(**[数据集](https://www.kaggle.com/vanguarde/h-m-eda-first-look)**)去训练一个BERT4Rec模型，但大致看了paper后，我不确定在这边是不是可以加上.item和.user两个atomic file去训练。所以想请教一下，对于一个sequential model我们怎么判断可以加哪些atomic file呢？谢谢！

**Comment by hyp1231** [comment, marked as answer]
您好！

我觉得可以分为两个层面来说：
1）框架层面。对于 RecBole 框架，加载哪些 atomic files 只取决于您的参数配置（可以传入额外 .yaml 来指定新的配置），加载文件时会按照参数配置加载指定的 atomic files，这些特征会存在 interaction 这个变量中作为模型的 calculate_loss, predict 等函数的输入。
2）模型层面。而模型是否要使用这些输入进来的特征，是由模型的代码确定的。例如 BERT4Rec 是纯粹基于 item 点击历史进行训练的模型，就不会使用 .item 和 .user 里的特征。

总结下来的话，您可以通过合适的参数配置，让 RecBole 加载 .user 和 .item 文件，但是 BERT4Rec 并不会使用他们。

**Reply by hlyu9** [reply]
好的，谢谢大佬！请问模型层面有什么比较快捷的判断方法吗，比如我看DIEN好像有取user和item，是不是可以认为会用到这两个atomic file？主要是想找一个可以用到user和item的sequence model，不知道是否可以请大佬建议一下…

另外想请教一下，sequence model是否支持用.ent的atomic file来自己加pretrained embedding呢？谢谢！

**Reply by hyp1231** [reply]
如果在 sequential recommendation 这个范畴内讨论的话，可以考虑 SASRecF, GRU4RecF, FDSA, S3Rec 等模型，这些都是加入 item feature 进行训练的模型；如果在 context-aware 这个范畴内，确实可以考虑 DIN 或者 DIEN；

可以的。

**Reply by hlyu9** [reply]
好的，谢谢大佬的指点！
